lib/flutter_radio.dart

```diff
@@ -1,36 +1,176 @@
 import 'dart:async';
-
+import 'dart:core';
+import 'dart:convert';
 import 'package:flutter/foundation.dart';
 import 'package:flutter/services.dart';
 
 class FlutterRadio {
   static const MethodChannel _channel = const MethodChannel('flutter_radio');
+  static StreamController<PlayStatus> _playerController;
+  /// Value ranges from 0 to 120
+  Stream<PlayStatus> get onPlayerStateChanged => _playerController.stream;
+
+  static bool _isPlaying = false;
+
+  static Future<void> audioStart(AudioPlayerItem item) async {
+    await FlutterRadio.setMeta(item);
+  }
 
-  static Future<void> audioStart() async {
-    await _channel.invokeMethod('audioStart');
+  static Future<void> playOrPause({@required String url}) async {
+    try {
+      if (FlutterRadio._isPlaying) {
+        FlutterRadio.pause(url: url);
+      } else {
+        FlutterRadio.play(url: url);
+      }
+    } catch (err) {
+      throw Exception(err);
+    }
   }
 
   static Future<void> play({@required String url}) async {
-    final Map<String, dynamic> params = <String, dynamic>{'url': url};
-    await _channel.invokeMethod('play', params);
+    try {
+      String result =
+          await _channel.invokeMethod('play', <String, dynamic>{
+        'path': url,
+      });
+      print('result: $result');
+
+      _setPlayerCallback();
+
+      if (FlutterRadio._isPlaying) {
+        throw Exception('Player is already playing.');
+      }
+      FlutterRadio._isPlaying = true;
+
+      return result;
+    } catch (err) {
+      throw Exception(err);
+    }
   }
 
   static Future<void> pause({@required String url}) async {
-    final Map<String, dynamic> params = <String, dynamic>{'url': url};
-    await _channel.invokeMethod('pause', params);
+    if (!FlutterRadio._isPlaying) {
+      throw Exception('Player already stopped.');
+    }
+    FlutterRadio._isPlaying = false;
+
+    String result = await _channel.invokeMethod('pause');
+    _removePlayerCallback();
+    return result;
   }
 
   static Future<void> stop() async {
-    await _channel.invokeMethod('stop');
+    if (!FlutterRadio._isPlaying) {
+      throw Exception('Player already stopped.');
+    }
+    FlutterRadio._isPlaying = false;
+
+    String result = await _channel.invokeMethod('stop');
+    _removePlayerCallback();
+    return result;
   }
 
   static Future<bool> isPlaying() async {
-    final bool playing = await _channel.invokeMethod('isPlaying');
-    return playing;
+    return Future.value(_isPlaying);
```

Comment:
tbh Future version of `isPlaying` is unusable. You cant observe it and change UI instantly when in changed. I put it for backward compatibility, but also stateChanged stream is imlemented.